lib/trento/application/integration/discovery/policies/cluster_policy.ex

```diff
@@ -1,140 +1,113 @@
-    do: nil
-
-  defp parse_cluster_sid(%{"Cib" => %{"Configuration" => %{"Resources" => %{"Clones" => clones}}}}) do
-    clones
-    |> Enum.find_value([], fn
-      %{"Primitive" => %{"Type" => "SAPHanaTopology", "InstanceAttributes" => attributes}} ->
-        attributes
-
-      _ ->
-        nil
-    end)
-    |> Enum.find_value(nil, fn
-      %{"Name" => "SID", "Value" => value} when value != "" ->
-        value
+  defp parse_resources_number(%{
+         crmmon: %{summary: %{resources: %{number: resources_number}}}
+       }),
+       do: resources_number
 
-      _ ->
-        nil
-    end)
-  end
+  defp parse_hosts_number(%{
+         crmmon: %{summary: %{nodes: %{number: nodes_number}}}
+       }),
+       do: nodes_number
 
   defp parse_cluster_details(
-         %{"Crmmon" => crmmon, "SBD" => sbd} = payload,
-         cluster_type,
-         sid
+         %{crmmon: crmmon, sbd: sbd, cluster_type: cluster_type, sid: sid} = payload
        )
        when cluster_type in [:hana_scale_up, :hana_scale_out] do
     nodes = parse_cluster_nodes(payload, sid)
 
     %{
-      "system_replication_mode" => parse_system_replication_mode(nodes, sid),
-      "system_replication_operation_mode" => parse_system_replication_operation_mode(nodes, sid),
-      "secondary_sync_state" => parse_secondary_sync_state(nodes, sid),
-      "sr_health_state" => parse_hana_sr_health_state(nodes, sid),
-      "fencing_type" => parse_cluster_fencing_type(crmmon),
-      "stopped_resources" => parse_cluster_stopped_resources(crmmon),
-      "nodes" => nodes,
-      "sbd_devices" => parse_sbd_devices(sbd)
+      system_replication_mode: parse_system_replication_mode(nodes, sid),
+      system_replication_operation_mode: parse_system_replication_operation_mode(nodes, sid),
+      secondary_sync_state: parse_secondary_sync_state(nodes, sid),
+      sr_health_state: parse_hana_sr_health_state(nodes, sid),
+      fencing_type: parse_cluster_fencing_type(crmmon),
+      stopped_resources: parse_cluster_stopped_resources(crmmon),
+      nodes: nodes,
+      sbd_devices: parse_sbd_devices(sbd)
     }
   end
 
-  defp parse_cluster_details(_, _, _), do: nil
+  defp parse_cluster_details(_) do
+    nil
+  end
 
   defp parse_cluster_nodes(
          %{
-           "Provider" => provider,
-           "Cib" => %{
-             "Configuration" => %{
-               "Resources" => resources
+           provider: provider,
+           cib: %{
+             configuration: %{
+               resources: resources
              }
            },
-           "Crmmon" =>
+           crmmon:
              %{
-               "NodeAttributes" => %{
-                 "Nodes" => nodes
+               node_attributes: %{
+                 nodes: nodes
                }
              } = crmmon
          },
          sid
        ) do
     nodes
-    |> to_list
-    |> Enum.map(fn %{"Name" => name, "Attributes" => attributes} ->
+    |> ListHelper.to_list()
```

Comment:
Do we need to still apply this `to_list` here? I thought we already sanitazed the data in the payload.
Finding `to_list` elements here is a smell to me